{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4aa52f9c-7028-4e20-9647-20d404654d3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import pytorch libraries\n",
    "import torch\n",
    "import torch.nn.functional as F\n",
    "from torch_geometric.nn import GCNConv\n",
    "from torch_geometric.nn import GATConv\n",
    "from torch_geometric.data import Data\n",
    "from torch_geometric.deprecation import deprecated\n",
    "\n",
    "import os\n",
    "import csv\n",
    "import copy\n",
    "import math\n",
    "import time\n",
    "import random\n",
    "import argparse\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import statsmodels.api as sm\n",
    "from statsmodels.tools.eval_measures import mse\n",
    "from decimal import Decimal\n",
    "from scipy import stats\n",
    "from scipy.stats import spearmanr, rankdata, norm\n",
    "from scipy.spatial.distance import cosine\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4f4dcbaa-c357-4328-8638-7045d5b975ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load normalized gtex data\n",
    "data_dir = '/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/12052023/'\n",
    "gt = pd.read_csv(data_dir+\"new_normed_gtex_gtex_allen_gene.txt\", low_memory=False, index_col=0, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5c770e7b-f1ed-43b8-9768-7dbc70e7f56a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GTEX-1192X-0011-R10a-SM-DO941</th>\n",
       "      <th>GTEX-1192X-0011-R5a-SM-DNZZA</th>\n",
       "      <th>GTEX-1192X-0011-R6a-SM-DNZZB</th>\n",
       "      <th>GTEX-1192X-0011-R7b-SM-DNZZC</th>\n",
       "      <th>GTEX-1192X-0011-R8b-SM-DO11U</th>\n",
       "      <th>GTEX-11DXW-0011-R1a-SM-DNZZD</th>\n",
       "      <th>GTEX-11DXW-0011-R3b-SM-DNZZE</th>\n",
       "      <th>GTEX-11DXW-0011-R5b-SM-DO11D</th>\n",
       "      <th>GTEX-11DXW-0011-R6a-SM-DO11N</th>\n",
       "      <th>GTEX-11DXW-0011-R7b-SM-DO119</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-ZVZQ-0011-R4b-SM-57WCZ</th>\n",
       "      <th>GTEX-ZVZQ-0011-R5b-SM-57WDC</th>\n",
       "      <th>GTEX-ZVZQ-0011-R6b-SM-57WAY</th>\n",
       "      <th>GTEX-ZVZQ-0011-R7b-SM-57WBB</th>\n",
       "      <th>GTEX-ZVZQ-0011-R8a-SM-51MR5</th>\n",
       "      <th>GTEX-ZXG5-0011-R10a-SM-57WDD</th>\n",
       "      <th>GTEX-ZXG5-0011-R5b-SM-57WBN</th>\n",
       "      <th>GTEX-ZXG5-0011-R6a-SM-57WBZ</th>\n",
       "      <th>GTEX-ZXG5-0011-R7b-SM-57WCC</th>\n",
       "      <th>GTEX-ZXG5-0011-R8a-SM-57WD1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>region</th>\n",
       "      <td>Frontal_Cortex_BA9</td>\n",
       "      <td>Caudate_basal_ganglia</td>\n",
       "      <td>Nucleus_accumbens_basal_ganglia</td>\n",
       "      <td>Putamen_basal_ganglia</td>\n",
       "      <td>Hypothalamus</td>\n",
       "      <td>Hippocampus</td>\n",
       "      <td>Anterior_cingulate_cortex_BA24</td>\n",
       "      <td>Caudate_basal_ganglia</td>\n",
       "      <td>Nucleus_accumbens_basal_ganglia</td>\n",
       "      <td>Putamen_basal_ganglia</td>\n",
       "      <td>...</td>\n",
       "      <td>Amygdala</td>\n",
       "      <td>Caudate_basal_ganglia</td>\n",
       "      <td>Nucleus_accumbens_basal_ganglia</td>\n",
       "      <td>Putamen_basal_ganglia</td>\n",
       "      <td>Hypothalamus</td>\n",
       "      <td>Frontal_Cortex_BA9</td>\n",
       "      <td>Caudate_basal_ganglia</td>\n",
       "      <td>Nucleus_accumbens_basal_ganglia</td>\n",
       "      <td>Putamen_basal_ganglia</td>\n",
       "      <td>Hypothalamus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>subject</th>\n",
       "      <td>GTEX-1192X</td>\n",
       "      <td>GTEX-1192X</td>\n",
       "      <td>GTEX-1192X</td>\n",
       "      <td>GTEX-1192X</td>\n",
       "      <td>GTEX-1192X</td>\n",
       "      <td>GTEX-11DXW</td>\n",
       "      <td>GTEX-11DXW</td>\n",
       "      <td>GTEX-11DXW</td>\n",
       "      <td>GTEX-11DXW</td>\n",
       "      <td>GTEX-11DXW</td>\n",
       "      <td>...</td>\n",
       "      <td>GTEX-ZVZQ</td>\n",
       "      <td>GTEX-ZVZQ</td>\n",
       "      <td>GTEX-ZVZQ</td>\n",
       "      <td>GTEX-ZVZQ</td>\n",
       "      <td>GTEX-ZVZQ</td>\n",
       "      <td>GTEX-ZXG5</td>\n",
       "      <td>GTEX-ZXG5</td>\n",
       "      <td>GTEX-ZXG5</td>\n",
       "      <td>GTEX-ZXG5</td>\n",
       "      <td>GTEX-ZXG5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-1.0609065014249923</td>\n",
       "      <td>0.1264239078327331</td>\n",
       "      <td>0.5148757621715246</td>\n",
       "      <td>0.588289446331391</td>\n",
       "      <td>-1.771613407678612</td>\n",
       "      <td>1.6978439236413676</td>\n",
       "      <td>0.3932091801194246</td>\n",
       "      <td>1.4801508340383251</td>\n",
       "      <td>-0.2147273736360117</td>\n",
       "      <td>1.5952158126759914</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5163158858752855</td>\n",
       "      <td>-0.3246145193310041</td>\n",
       "      <td>-0.5091258922853752</td>\n",
       "      <td>1.176396124442838</td>\n",
       "      <td>-0.8885490700739215</td>\n",
       "      <td>-0.2614278589963243</td>\n",
       "      <td>-0.123882321258613</td>\n",
       "      <td>1.0092030513011092</td>\n",
       "      <td>0.0327887390146478</td>\n",
       "      <td>-0.4777940721655931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.3259438994702136</td>\n",
       "      <td>0.6634502165382117</td>\n",
       "      <td>-0.3823344838209241</td>\n",
       "      <td>0.8866794429910266</td>\n",
       "      <td>0.3918473238070559</td>\n",
       "      <td>-0.6462636078650708</td>\n",
       "      <td>-0.9363469789541222</td>\n",
       "      <td>-0.2380127215108003</td>\n",
       "      <td>-0.7592782647155094</td>\n",
       "      <td>1.0008457065936749</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0390972201647005</td>\n",
       "      <td>0.1493369929874266</td>\n",
       "      <td>-0.6888055000021085</td>\n",
       "      <td>0.2289426628534391</td>\n",
       "      <td>-0.3986639351466136</td>\n",
       "      <td>-0.722773723295188</td>\n",
       "      <td>-0.2018420057931938</td>\n",
       "      <td>-0.6109276409054578</td>\n",
       "      <td>0.4013957572006627</td>\n",
       "      <td>0.4441396106266967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>0.5322289934322247</td>\n",
       "      <td>1.9928519061306584</td>\n",
       "      <td>0.7326323477704342</td>\n",
       "      <td>1.95738848854114</td>\n",
       "      <td>-0.3193027047079684</td>\n",
       "      <td>0.3021004981385619</td>\n",
       "      <td>0.6634502165382117</td>\n",
       "      <td>0.9343941706335293</td>\n",
       "      <td>-0.3755607691893882</td>\n",
       "      <td>0.4497128493832243</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8979442131538682</td>\n",
       "      <td>1.6088649489713986</td>\n",
       "      <td>1.1226894351665453</td>\n",
       "      <td>-0.1353260055415081</td>\n",
       "      <td>0.1748888113730733</td>\n",
       "      <td>0.8960588596621761</td>\n",
       "      <td>1.3161348891608369</td>\n",
       "      <td>0.4385801333361819</td>\n",
       "      <td>1.094714233035716</td>\n",
       "      <td>-0.1340536369893952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDC</th>\n",
       "      <td>-1.4992730266925278</td>\n",
       "      <td>-0.2994620483153727</td>\n",
       "      <td>-0.4000294730034018</td>\n",
       "      <td>0.2173086623150939</td>\n",
       "      <td>-0.591290258230665</td>\n",
       "      <td>-0.1455130221226983</td>\n",
       "      <td>-0.6033473103603407</td>\n",
       "      <td>0.8663141521808834</td>\n",
       "      <td>0.3715048161950432</td>\n",
       "      <td>0.8553559451025562</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.2849872324504064</td>\n",
       "      <td>0.5512111556884931</td>\n",
       "      <td>0.5867910262845815</td>\n",
       "      <td>0.9701094246784554</td>\n",
       "      <td>0.0226977782392312</td>\n",
       "      <td>-1.2283492364529145</td>\n",
       "      <td>0.1327814854227067</td>\n",
       "      <td>0.0656127766985439</td>\n",
       "      <td>0.8810890606408079</td>\n",
       "      <td>-0.5988157254156958</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>1.8027430907391908</td>\n",
       "      <td>-0.2863006147481918</td>\n",
       "      <td>-0.0353119608314343</td>\n",
       "      <td>-0.4806226588755429</td>\n",
       "      <td>1.5731064789417957</td>\n",
       "      <td>-0.1416911778834665</td>\n",
       "      <td>0.5748505821262617</td>\n",
       "      <td>-0.9188980626667594</td>\n",
       "      <td>0.5177570811930351</td>\n",
       "      <td>-1.4293018134072648</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1966973470096366</td>\n",
       "      <td>-0.8055139860387096</td>\n",
       "      <td>-0.0138701182615907</td>\n",
       "      <td>-0.964071539514268</td>\n",
       "      <td>0.6063752591849039</td>\n",
       "      <td>0.5497440049149145</td>\n",
       "      <td>-0.8283988316306032</td>\n",
       "      <td>-0.2902437364955883</td>\n",
       "      <td>-0.9660802544456248</td>\n",
       "      <td>0.6744897501960817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>0.0454072576636208</td>\n",
       "      <td>-1.577466916200561</td>\n",
       "      <td>-1.127437990992025</td>\n",
       "      <td>-1.1942137490551366</td>\n",
       "      <td>-0.7409025733519962</td>\n",
       "      <td>-0.964071539514268</td>\n",
       "      <td>0.6018354095723709</td>\n",
       "      <td>-0.092174552477814</td>\n",
       "      <td>0.020175439267082</td>\n",
       "      <td>0.1531631488259466</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.5453496267474041</td>\n",
       "      <td>-0.6665960684308483</td>\n",
       "      <td>-0.5748505821262617</td>\n",
       "      <td>-1.1763961244428374</td>\n",
       "      <td>-0.1149929573606819</td>\n",
       "      <td>0.4623042126606105</td>\n",
       "      <td>-1.1814486345934392</td>\n",
       "      <td>-0.1506121326041637</td>\n",
       "      <td>-1.0050156044940677</td>\n",
       "      <td>-0.2575157167511276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>-1.2364326481125962</td>\n",
       "      <td>-0.7796215252577617</td>\n",
       "      <td>-0.8755260797151984</td>\n",
       "      <td>-0.0567700523402176</td>\n",
       "      <td>-0.3539984561420949</td>\n",
       "      <td>-0.6650223198196294</td>\n",
       "      <td>-0.0934408738335693</td>\n",
       "      <td>0.3219574764987721</td>\n",
       "      <td>1.0092030513011092</td>\n",
       "      <td>0.4637077514571792</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.3877660998272049</td>\n",
       "      <td>0.1315095487093334</td>\n",
       "      <td>-0.9782153889713696</td>\n",
       "      <td>0.0731963141600561</td>\n",
       "      <td>-0.3432793291935854</td>\n",
       "      <td>-0.8792317108840424</td>\n",
       "      <td>0.0921745524778142</td>\n",
       "      <td>0.1124548379103731</td>\n",
       "      <td>0.0833143256722081</td>\n",
       "      <td>0.1353260055415081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>0.4679238681014364</td>\n",
       "      <td>-0.573363807605522</td>\n",
       "      <td>-0.1429648932640479</td>\n",
       "      <td>0.3620645112555753</td>\n",
       "      <td>1.127437990992025</td>\n",
       "      <td>0.4137264599896821</td>\n",
       "      <td>-0.5718782994258718</td>\n",
       "      <td>-0.4123533057440129</td>\n",
       "      <td>0.2889288641259237</td>\n",
       "      <td>0.598815725415696</td>\n",
       "      <td>...</td>\n",
       "      <td>0.6888055000021083</td>\n",
       "      <td>-0.2679570303334746</td>\n",
       "      <td>0.1111860508200961</td>\n",
       "      <td>2.011567070291372</td>\n",
       "      <td>0.3850488718359516</td>\n",
       "      <td>0.7048793663367097</td>\n",
       "      <td>0.4096093250145137</td>\n",
       "      <td>-0.1137238060586805</td>\n",
       "      <td>1.3786326378189782</td>\n",
       "      <td>0.9782153889713696</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15046 rows × 1987 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GTEX-1192X-0011-R10a-SM-DO941 GTEX-1192X-0011-R5a-SM-DNZZA  \\\n",
       "region             Frontal_Cortex_BA9        Caudate_basal_ganglia   \n",
       "subject                    GTEX-1192X                   GTEX-1192X   \n",
       "A1BG              -1.0609065014249923           0.1264239078327331   \n",
       "A2M               -0.3259438994702136           0.6634502165382117   \n",
       "A2ML1              0.5322289934322247           1.9928519061306584   \n",
       "...                               ...                          ...   \n",
       "ZXDC              -1.4992730266925278          -0.2994620483153727   \n",
       "ZYG11B             1.8027430907391908          -0.2863006147481918   \n",
       "ZYX                0.0454072576636208           -1.577466916200561   \n",
       "ZZEF1             -1.2364326481125962          -0.7796215252577617   \n",
       "ZZZ3               0.4679238681014364           -0.573363807605522   \n",
       "\n",
       "            GTEX-1192X-0011-R6a-SM-DNZZB GTEX-1192X-0011-R7b-SM-DNZZC  \\\n",
       "region   Nucleus_accumbens_basal_ganglia        Putamen_basal_ganglia   \n",
       "subject                       GTEX-1192X                   GTEX-1192X   \n",
       "A1BG                  0.5148757621715246            0.588289446331391   \n",
       "A2M                  -0.3823344838209241           0.8866794429910266   \n",
       "A2ML1                 0.7326323477704342             1.95738848854114   \n",
       "...                                  ...                          ...   \n",
       "ZXDC                 -0.4000294730034018           0.2173086623150939   \n",
       "ZYG11B               -0.0353119608314343          -0.4806226588755429   \n",
       "ZYX                   -1.127437990992025          -1.1942137490551366   \n",
       "ZZEF1                -0.8755260797151984          -0.0567700523402176   \n",
       "ZZZ3                 -0.1429648932640479           0.3620645112555753   \n",
       "\n",
       "        GTEX-1192X-0011-R8b-SM-DO11U GTEX-11DXW-0011-R1a-SM-DNZZD  \\\n",
       "region                  Hypothalamus                  Hippocampus   \n",
       "subject                   GTEX-1192X                   GTEX-11DXW   \n",
       "A1BG              -1.771613407678612           1.6978439236413676   \n",
       "A2M               0.3918473238070559          -0.6462636078650708   \n",
       "A2ML1            -0.3193027047079684           0.3021004981385619   \n",
       "...                              ...                          ...   \n",
       "ZXDC              -0.591290258230665          -0.1455130221226983   \n",
       "ZYG11B            1.5731064789417957          -0.1416911778834665   \n",
       "ZYX              -0.7409025733519962           -0.964071539514268   \n",
       "ZZEF1            -0.3539984561420949          -0.6650223198196294   \n",
       "ZZZ3               1.127437990992025           0.4137264599896821   \n",
       "\n",
       "           GTEX-11DXW-0011-R3b-SM-DNZZE GTEX-11DXW-0011-R5b-SM-DO11D  \\\n",
       "region   Anterior_cingulate_cortex_BA24        Caudate_basal_ganglia   \n",
       "subject                      GTEX-11DXW                   GTEX-11DXW   \n",
       "A1BG                 0.3932091801194246           1.4801508340383251   \n",
       "A2M                 -0.9363469789541222          -0.2380127215108003   \n",
       "A2ML1                0.6634502165382117           0.9343941706335293   \n",
       "...                                 ...                          ...   \n",
       "ZXDC                -0.6033473103603407           0.8663141521808834   \n",
       "ZYG11B               0.5748505821262617          -0.9188980626667594   \n",
       "ZYX                  0.6018354095723709           -0.092174552477814   \n",
       "ZZEF1               -0.0934408738335693           0.3219574764987721   \n",
       "ZZZ3                -0.5718782994258718          -0.4123533057440129   \n",
       "\n",
       "            GTEX-11DXW-0011-R6a-SM-DO11N GTEX-11DXW-0011-R7b-SM-DO119  ...  \\\n",
       "region   Nucleus_accumbens_basal_ganglia        Putamen_basal_ganglia  ...   \n",
       "subject                       GTEX-11DXW                   GTEX-11DXW  ...   \n",
       "A1BG                 -0.2147273736360117           1.5952158126759914  ...   \n",
       "A2M                  -0.7592782647155094           1.0008457065936749  ...   \n",
       "A2ML1                -0.3755607691893882           0.4497128493832243  ...   \n",
       "...                                  ...                          ...  ...   \n",
       "ZXDC                  0.3715048161950432           0.8553559451025562  ...   \n",
       "ZYG11B                0.5177570811930351          -1.4293018134072648  ...   \n",
       "ZYX                    0.020175439267082           0.1531631488259466  ...   \n",
       "ZZEF1                 1.0092030513011092           0.4637077514571792  ...   \n",
       "ZZZ3                  0.2889288641259237            0.598815725415696  ...   \n",
       "\n",
       "        GTEX-ZVZQ-0011-R4b-SM-57WCZ GTEX-ZVZQ-0011-R5b-SM-57WDC  \\\n",
       "region                     Amygdala       Caudate_basal_ganglia   \n",
       "subject                   GTEX-ZVZQ                   GTEX-ZVZQ   \n",
       "A1BG             0.5163158858752855         -0.3246145193310041   \n",
       "A2M              0.0390972201647005          0.1493369929874266   \n",
       "A2ML1            0.8979442131538682          1.6088649489713986   \n",
       "...                             ...                         ...   \n",
       "ZXDC            -0.2849872324504064          0.5512111556884931   \n",
       "ZYG11B           0.1966973470096366         -0.8055139860387096   \n",
       "ZYX             -0.5453496267474041         -0.6665960684308483   \n",
       "ZZEF1           -0.3877660998272049          0.1315095487093334   \n",
       "ZZZ3             0.6888055000021083         -0.2679570303334746   \n",
       "\n",
       "             GTEX-ZVZQ-0011-R6b-SM-57WAY GTEX-ZVZQ-0011-R7b-SM-57WBB  \\\n",
       "region   Nucleus_accumbens_basal_ganglia       Putamen_basal_ganglia   \n",
       "subject                        GTEX-ZVZQ                   GTEX-ZVZQ   \n",
       "A1BG                 -0.5091258922853752           1.176396124442838   \n",
       "A2M                  -0.6888055000021085          0.2289426628534391   \n",
       "A2ML1                 1.1226894351665453         -0.1353260055415081   \n",
       "...                                  ...                         ...   \n",
       "ZXDC                  0.5867910262845815          0.9701094246784554   \n",
       "ZYG11B               -0.0138701182615907          -0.964071539514268   \n",
       "ZYX                  -0.5748505821262617         -1.1763961244428374   \n",
       "ZZEF1                -0.9782153889713696          0.0731963141600561   \n",
       "ZZZ3                  0.1111860508200961           2.011567070291372   \n",
       "\n",
       "        GTEX-ZVZQ-0011-R8a-SM-51MR5 GTEX-ZXG5-0011-R10a-SM-57WDD  \\\n",
       "region                 Hypothalamus           Frontal_Cortex_BA9   \n",
       "subject                   GTEX-ZVZQ                    GTEX-ZXG5   \n",
       "A1BG            -0.8885490700739215          -0.2614278589963243   \n",
       "A2M             -0.3986639351466136           -0.722773723295188   \n",
       "A2ML1            0.1748888113730733           0.8960588596621761   \n",
       "...                             ...                          ...   \n",
       "ZXDC             0.0226977782392312          -1.2283492364529145   \n",
       "ZYG11B           0.6063752591849039           0.5497440049149145   \n",
       "ZYX             -0.1149929573606819           0.4623042126606105   \n",
       "ZZEF1           -0.3432793291935854          -0.8792317108840424   \n",
       "ZZZ3             0.3850488718359516           0.7048793663367097   \n",
       "\n",
       "        GTEX-ZXG5-0011-R5b-SM-57WBN      GTEX-ZXG5-0011-R6a-SM-57WBZ  \\\n",
       "region        Caudate_basal_ganglia  Nucleus_accumbens_basal_ganglia   \n",
       "subject                   GTEX-ZXG5                        GTEX-ZXG5   \n",
       "A1BG             -0.123882321258613               1.0092030513011092   \n",
       "A2M             -0.2018420057931938              -0.6109276409054578   \n",
       "A2ML1            1.3161348891608369               0.4385801333361819   \n",
       "...                             ...                              ...   \n",
       "ZXDC             0.1327814854227067               0.0656127766985439   \n",
       "ZYG11B          -0.8283988316306032              -0.2902437364955883   \n",
       "ZYX             -1.1814486345934392              -0.1506121326041637   \n",
       "ZZEF1            0.0921745524778142               0.1124548379103731   \n",
       "ZZZ3             0.4096093250145137              -0.1137238060586805   \n",
       "\n",
       "        GTEX-ZXG5-0011-R7b-SM-57WCC GTEX-ZXG5-0011-R8a-SM-57WD1  \n",
       "region        Putamen_basal_ganglia                Hypothalamus  \n",
       "subject                   GTEX-ZXG5                   GTEX-ZXG5  \n",
       "A1BG             0.0327887390146478         -0.4777940721655931  \n",
       "A2M              0.4013957572006627          0.4441396106266967  \n",
       "A2ML1             1.094714233035716         -0.1340536369893952  \n",
       "...                             ...                         ...  \n",
       "ZXDC             0.8810890606408079         -0.5988157254156958  \n",
       "ZYG11B          -0.9660802544456248          0.6744897501960817  \n",
       "ZYX             -1.0050156044940677         -0.2575157167511276  \n",
       "ZZEF1            0.0833143256722081          0.1353260055415081  \n",
       "ZZZ3             1.3786326378189782          0.9782153889713696  \n",
       "\n",
       "[15046 rows x 1987 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f379e408-f416-489b-97f3-47017f47d7d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define a function to apply inverse normal transformation on a pandas Series\n",
    "def inverse_normal_transform(series):\n",
    "    # Rank the data (average rank for ties)\n",
    "    ranks = rankdata(series)\n",
    "    # Convert ranks to uniform values between (0, 1) by dividing by length+1\n",
    "    uniform = (ranks - 0.5) / len(ranks)\n",
    "    # Apply the inverse normal transformation\n",
    "    return pd.Series(norm.ppf(uniform))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "57d53b6c-90cf-43b1-a77c-fb7bd63adb1e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# region_pick = ['Amygdala', 'Anterior_cingulate_cortex_BA24', 'Caudate_basal_ganglia', \n",
    "#                'Cerebellar_Hemisphere', 'Frontal_Cortex_BA9', 'Hippocampus', 'Hypothalamus', \n",
    "#                'Nucleus_accumbens_basal_ganglia', 'Putamen_basal_ganglia', 'Substantia_nigra']\n",
    "\n",
    "# pred_region = region_pick[0]\n",
    "# t_epoch = 300\n",
    "# f_epoch = 500\n",
    "# missing_N = 5\n",
    "# save_dir = '/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/02162024/ATG_8_or_less/Prediction/'\n",
    "# csv_file = f'{pred_region}_{t_epoch}_{f_epoch}_{missing_N}_LMfromGTEx.csv'\n",
    "# go_mat = pd.read_csv(save_dir+csv_file, index_col=0)\n",
    "\n",
    "# # data = go_mat.iloc[0, :].tolist()\n",
    "# # plt.hist(data, bins=5, edgecolor='black')\n",
    "\n",
    "# # Apply the function across the DataFrame row-wise\n",
    "# transformed_df = go_mat.apply(inverse_normal_transform, axis=1)\n",
    "# transformed_df.index = go_mat.index\n",
    "# transformed_df.columns = go_mat.columns\n",
    "\n",
    "# # data = transformed_df.iloc[0, :].tolist()\n",
    "# # plt.hist(data, bins=5, edgecolor='black')\n",
    "\n",
    "# transformed_df.reset_index(inplace=True)\n",
    "# transformed_df.rename(columns={'index': 'Gene'}, inplace=True)\n",
    "\n",
    "# # exp_mat_dir = '/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/02162024/eqtl_analysis/GTEx_Analysis_v8_eQTL_expression_matrices/'\n",
    "# # sub_gt_filename = 'Brain_Amygdala.v8.normalized_expression.bed.gz'\n",
    "# match_file = '/project/pi_rachel_melamed_uml_edu/adni/data/ensg2gene_name.txt'\n",
    "# ensg2gene = pd.read_csv(match_file, sep='\\t')\n",
    "# all_bedgz = pd.read_table('/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/09032023/Gtex_from_raw_to_normalized/normed_all_10_brain_tissues.expression.bed.gz')\n",
    "# gene_list = [ensg2gene['gene_name'][ensg2gene['Unnamed: 0'].tolist().index(all_bedgz['gene_id'][i])] for i in range(all_bedgz.shape[0])]\n",
    "# gene_name = pd.Series(gene_list, name='Gene')\n",
    "# mat = pd.concat([all_bedgz.iloc[:, :4], gene_name], axis=1)\n",
    "# # Perform a left merge on 'RowName'\n",
    "# mat_merge = pd.merge(mat, transformed_df, on='Gene', how='left')\n",
    "# mat = mat_merge.dropna().reset_index(drop=True)\n",
    "# # remove duplicated genes\n",
    "# duplicated_row = mat.duplicated(subset='Gene', keep='first')\n",
    "# final_mat = mat[~duplicated_row].reset_index(drop=True)\n",
    "# # remove the 'Gene' column\n",
    "# final_mat.drop('Gene', axis=1, inplace=True)\n",
    "\n",
    "# save_dir = '/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/02162024/eqtl_analysis/new_bedgz_file/'\n",
    "# filename = f'{pred_region}_newnormed.expression.bed.gz'\n",
    "# final_mat.to_csv(save_dir+filename, sep='\\t', index=False, header=True, compression='gzip')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8b4ddeda-c334-4272-9161-1a6495b62889",
   "metadata": {},
   "outputs": [],
   "source": [
    "# concat PCs from all the tissue.txt file\n",
    "data_dir = '/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/02162024/eqtl_analysis/GTEx_Analysis_v8_eQTL_covariates/'\n",
    "data_dict = {}\n",
    "files = os.listdir(data_dir)\n",
    "# read all the covariate files from all the tissues\n",
    "for file in files:\n",
    "    if file.endswith('.v8.covariates.txt'):\n",
    "        prefix = file.split('.v8.covariates.txt')[0]\n",
    "        df = pd.read_csv(os.path.join(data_dir, file), low_memory=False, index_col=0, sep=\"\\t\")\n",
    "        PCs = df.iloc[:5]\n",
    "        others = df.iloc[-3:]\n",
    "        df_sub = pd.concat([PCs, others])\n",
    "        data_dict[prefix] = df_sub\n",
    "# build a matrix with all the subjects\n",
    "combined_df = pd.concat(data_dict.values(), axis=1)\n",
    "combined_df = combined_df.loc[:, ~combined_df.columns.duplicated()]\n",
    "combined_df = combined_df.reindex(sorted(combined_df.columns), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a7ec19c6-0207-42f1-ae83-b75691572d54",
   "metadata": {},
   "outputs": [],
   "source": [
    "region_pick = ['Amygdala', 'Anterior_cingulate_cortex_BA24', 'Caudate_basal_ganglia', \n",
    "               'Cerebellar_Hemisphere', 'Frontal_Cortex_BA9', 'Hippocampus', 'Hypothalamus', \n",
    "               'Nucleus_accumbens_basal_ganglia', 'Putamen_basal_ganglia', 'Substantia_nigra']\n",
    "\n",
    "t_epoch = 300\n",
    "f_epoch = 500\n",
    "missing_N = 5\n",
    "\n",
    "for pred_region in region_pick:\n",
    "    save_dir = '/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/02162024/ATG_8_or_less/Prediction/'\n",
    "    csv_file = f'{pred_region}_{t_epoch}_{f_epoch}_{missing_N}_LMfromGTEx.csv'\n",
    "    go_mat = pd.read_csv(save_dir+csv_file, index_col=0)\n",
    "    \n",
    "    # only use the subjects that also have genotype data\n",
    "    overlapped_subjects = [x for x in combined_df.columns if x in go_mat.columns] \n",
    "    go_mat = go_mat[overlapped_subjects]\n",
    "    \n",
    "    # Apply the function across the DataFrame row-wise\n",
    "    transformed_df = go_mat.apply(inverse_normal_transform, axis=1)\n",
    "    transformed_df.index = go_mat.index\n",
    "    transformed_df.columns = go_mat.columns\n",
    "    # data = transformed_df.iloc[0, :].tolist()\n",
    "    # plt.hist(data, bins=5, edgecolor='black')\n",
    "    transformed_df.reset_index(inplace=True)\n",
    "    transformed_df.rename(columns={'index': 'Gene'}, inplace=True)\n",
    "    \n",
    "    # exp_mat_dir = '/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/02162024/eqtl_analysis/GTEx_Analysis_v8_eQTL_expression_matrices/'\n",
    "    # sub_gt_filename = 'Brain_Amygdala.v8.normalized_expression.bed.gz'\n",
    "    match_file = '/project/pi_rachel_melamed_uml_edu/adni/data/ensg2gene_name.txt'\n",
    "    ensg2gene = pd.read_csv(match_file, sep='\\t')\n",
    "    all_bedgz = pd.read_table('/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/09032023/Gtex_from_raw_to_normalized/normed_all_10_brain_tissues.expression.bed.gz')\n",
    "    gene_list = [ensg2gene['gene_name'][ensg2gene['Unnamed: 0'].tolist().index(all_bedgz['gene_id'][i])] for i in range(all_bedgz.shape[0])]\n",
    "    gene_name = pd.Series(gene_list, name='Gene')\n",
    "    mat = pd.concat([all_bedgz.iloc[:, :4], gene_name], axis=1)\n",
    "    # Perform a left merge on 'RowName'\n",
    "    mat_merge = pd.merge(mat, transformed_df, on='Gene', how='left')\n",
    "    mat = mat_merge.dropna().reset_index(drop=True)\n",
    "    # remove duplicated genes\n",
    "    duplicated_row = mat.duplicated(subset='Gene', keep='first')\n",
    "    final_mat = mat[~duplicated_row].reset_index(drop=True)\n",
    "    # remove the 'Gene' column\n",
    "    final_mat.drop('Gene', axis=1, inplace=True)\n",
    "    \n",
    "    # need to sorted the genes and also remove genes in chrY\n",
    "    # chr_order = ['chr1', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17', 'chr18', 'chr19', \n",
    "    #              'chr2', 'chr20', 'chr21', 'chr22', 'chr3', 'chr4', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9', 'chrX', 'chrY']\n",
    "    chr_order = ['chr1', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17', 'chr18', 'chr19', \n",
    "                 'chr2', 'chr20', 'chr21', 'chr22', 'chr3', 'chr4', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9', 'chrX']\n",
    "    new_row_order = []\n",
    "    chr_list = final_mat['#chr']\n",
    "    for chrom in chr_order:\n",
    "        for i in range(final_mat.shape[0]):\n",
    "            if chr_list[i]==chrom:\n",
    "                new_row_order.append(i)\n",
    "    sorted_final_mat = final_mat.loc[new_row_order]\n",
    "    \n",
    "    # save the result\n",
    "    save_dir = '/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/02162024/eqtl_analysis/new_bedgz_file/'\n",
    "    filename = f'{pred_region}_newnormed.expression.bed.gz'\n",
    "    sorted_final_mat.to_csv(save_dir+filename, sep='\\t', index=False, header=True, compression='gzip')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "02693d42-1ee7-48cc-a8cf-140247bee902",
   "metadata": {},
   "source": [
    "### See the file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d81c66b1-8319-4da9-957f-647ca201b181",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GTEX-1192X</th>\n",
       "      <th>GTEX-11DYG</th>\n",
       "      <th>GTEX-11DZ1</th>\n",
       "      <th>GTEX-11GSP</th>\n",
       "      <th>GTEX-11NUK</th>\n",
       "      <th>GTEX-11OF3</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-ZAB4</th>\n",
       "      <th>GTEX-ZAK1</th>\n",
       "      <th>GTEX-ZDXO</th>\n",
       "      <th>GTEX-ZE7O</th>\n",
       "      <th>GTEX-ZF28</th>\n",
       "      <th>GTEX-ZUA1</th>\n",
       "      <th>GTEX-ZV68</th>\n",
       "      <th>GTEX-ZVT3</th>\n",
       "      <th>GTEX-ZVZQ</th>\n",
       "      <th>GTEX-ZXG5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69090</td>\n",
       "      <td>69091</td>\n",
       "      <td>ENSG00000186092.4</td>\n",
       "      <td>-0.369009</td>\n",
       "      <td>-0.898936</td>\n",
       "      <td>-1.062786</td>\n",
       "      <td>-0.643034</td>\n",
       "      <td>0.382595</td>\n",
       "      <td>-0.423792</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.861634</td>\n",
       "      <td>-1.840761</td>\n",
       "      <td>-0.120559</td>\n",
       "      <td>0.288809</td>\n",
       "      <td>0.342037</td>\n",
       "      <td>-0.396252</td>\n",
       "      <td>0.355490</td>\n",
       "      <td>-0.044324</td>\n",
       "      <td>-0.807824</td>\n",
       "      <td>0.031655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827521</td>\n",
       "      <td>827522</td>\n",
       "      <td>ENSG00000225880.5</td>\n",
       "      <td>-0.249427</td>\n",
       "      <td>-1.350775</td>\n",
       "      <td>0.597003</td>\n",
       "      <td>-0.107816</td>\n",
       "      <td>0.031655</td>\n",
       "      <td>-0.082381</td>\n",
       "      <td>...</td>\n",
       "      <td>0.082381</td>\n",
       "      <td>0.146105</td>\n",
       "      <td>-0.627544</td>\n",
       "      <td>0.552205</td>\n",
       "      <td>1.416717</td>\n",
       "      <td>0.382595</td>\n",
       "      <td>0.369009</td>\n",
       "      <td>0.465723</td>\n",
       "      <td>-0.508488</td>\n",
       "      <td>-0.069685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>876753</td>\n",
       "      <td>876754</td>\n",
       "      <td>ENSG00000230368.2</td>\n",
       "      <td>-0.807824</td>\n",
       "      <td>-1.156505</td>\n",
       "      <td>0.315311</td>\n",
       "      <td>-0.006330</td>\n",
       "      <td>-1.207414</td>\n",
       "      <td>-0.031655</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.107816</td>\n",
       "      <td>-0.249427</td>\n",
       "      <td>0.018991</td>\n",
       "      <td>0.508488</td>\n",
       "      <td>0.262509</td>\n",
       "      <td>0.184609</td>\n",
       "      <td>0.612202</td>\n",
       "      <td>0.494133</td>\n",
       "      <td>1.181577</td>\n",
       "      <td>0.369009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>911434</td>\n",
       "      <td>911435</td>\n",
       "      <td>ENSG00000230699.2</td>\n",
       "      <td>-0.382595</td>\n",
       "      <td>0.918063</td>\n",
       "      <td>1.529007</td>\n",
       "      <td>-0.451658</td>\n",
       "      <td>-0.597003</td>\n",
       "      <td>0.706628</td>\n",
       "      <td>...</td>\n",
       "      <td>0.690469</td>\n",
       "      <td>-0.465723</td>\n",
       "      <td>0.998201</td>\n",
       "      <td>0.790402</td>\n",
       "      <td>1.717660</td>\n",
       "      <td>0.133321</td>\n",
       "      <td>0.288809</td>\n",
       "      <td>0.423792</td>\n",
       "      <td>0.722972</td>\n",
       "      <td>0.756257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923927</td>\n",
       "      <td>923928</td>\n",
       "      <td>ENSG00000187634.11</td>\n",
       "      <td>0.479879</td>\n",
       "      <td>0.581942</td>\n",
       "      <td>0.369009</td>\n",
       "      <td>0.756257</td>\n",
       "      <td>-2.428737</td>\n",
       "      <td>1.207414</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.508488</td>\n",
       "      <td>1.234084</td>\n",
       "      <td>-0.031655</td>\n",
       "      <td>0.977579</td>\n",
       "      <td>0.825494</td>\n",
       "      <td>0.522948</td>\n",
       "      <td>0.567010</td>\n",
       "      <td>0.658680</td>\n",
       "      <td>0.146105</td>\n",
       "      <td>0.612202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15023</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155197006</td>\n",
       "      <td>155197007</td>\n",
       "      <td>ENSG00000155959.10</td>\n",
       "      <td>0.120559</td>\n",
       "      <td>0.249427</td>\n",
       "      <td>-1.062786</td>\n",
       "      <td>-1.132140</td>\n",
       "      <td>1.234084</td>\n",
       "      <td>-1.156505</td>\n",
       "      <td>...</td>\n",
       "      <td>1.207414</td>\n",
       "      <td>-0.210428</td>\n",
       "      <td>-0.739512</td>\n",
       "      <td>-0.508488</td>\n",
       "      <td>-0.825494</td>\n",
       "      <td>0.275635</td>\n",
       "      <td>-0.018991</td>\n",
       "      <td>-0.756257</td>\n",
       "      <td>0.739512</td>\n",
       "      <td>-0.451658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15024</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155264588</td>\n",
       "      <td>155264589</td>\n",
       "      <td>ENSG00000155961.4</td>\n",
       "      <td>0.807824</td>\n",
       "      <td>1.062786</td>\n",
       "      <td>0.937532</td>\n",
       "      <td>-0.315311</td>\n",
       "      <td>1.452133</td>\n",
       "      <td>0.057000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.234084</td>\n",
       "      <td>-0.658680</td>\n",
       "      <td>-0.396252</td>\n",
       "      <td>-0.581942</td>\n",
       "      <td>-1.207414</td>\n",
       "      <td>-0.031655</td>\n",
       "      <td>-1.416717</td>\n",
       "      <td>-1.840761</td>\n",
       "      <td>0.627544</td>\n",
       "      <td>-0.236388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15025</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155334656</td>\n",
       "      <td>155334657</td>\n",
       "      <td>ENSG00000155962.12</td>\n",
       "      <td>-0.057000</td>\n",
       "      <td>-0.107816</td>\n",
       "      <td>0.018991</td>\n",
       "      <td>-0.210428</td>\n",
       "      <td>-1.181577</td>\n",
       "      <td>-0.275635</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.616156</td>\n",
       "      <td>-0.957363</td>\n",
       "      <td>-1.085325</td>\n",
       "      <td>-0.537519</td>\n",
       "      <td>1.108429</td>\n",
       "      <td>-0.158913</td>\n",
       "      <td>0.031655</td>\n",
       "      <td>0.249427</td>\n",
       "      <td>-0.522948</td>\n",
       "      <td>-0.184609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15026</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155669943</td>\n",
       "      <td>155669944</td>\n",
       "      <td>ENSG00000185973.10</td>\n",
       "      <td>0.451658</td>\n",
       "      <td>0.315311</td>\n",
       "      <td>-0.315311</td>\n",
       "      <td>-0.146105</td>\n",
       "      <td>1.181577</td>\n",
       "      <td>-0.184609</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.062786</td>\n",
       "      <td>0.581942</td>\n",
       "      <td>-0.197502</td>\n",
       "      <td>-1.914506</td>\n",
       "      <td>-0.937532</td>\n",
       "      <td>-0.107816</td>\n",
       "      <td>0.275635</td>\n",
       "      <td>0.044324</td>\n",
       "      <td>1.664765</td>\n",
       "      <td>-0.627544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15027</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155767811</td>\n",
       "      <td>155767812</td>\n",
       "      <td>ENSG00000168939.11</td>\n",
       "      <td>0.465723</td>\n",
       "      <td>0.107816</td>\n",
       "      <td>0.790402</td>\n",
       "      <td>-0.158913</td>\n",
       "      <td>1.019257</td>\n",
       "      <td>0.756257</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.552205</td>\n",
       "      <td>-1.452133</td>\n",
       "      <td>-0.275635</td>\n",
       "      <td>-0.302034</td>\n",
       "      <td>0.057000</td>\n",
       "      <td>0.522948</td>\n",
       "      <td>-0.465723</td>\n",
       "      <td>-0.095091</td>\n",
       "      <td>1.914506</td>\n",
       "      <td>0.328645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15028 rows × 202 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #chr      start        end             gene_id  GTEX-1192X  GTEX-11DYG  \\\n",
       "0      chr1      69090      69091   ENSG00000186092.4   -0.369009   -0.898936   \n",
       "1      chr1     827521     827522   ENSG00000225880.5   -0.249427   -1.350775   \n",
       "2      chr1     876753     876754   ENSG00000230368.2   -0.807824   -1.156505   \n",
       "3      chr1     911434     911435   ENSG00000230699.2   -0.382595    0.918063   \n",
       "4      chr1     923927     923928  ENSG00000187634.11    0.479879    0.581942   \n",
       "...     ...        ...        ...                 ...         ...         ...   \n",
       "15023  chrX  155197006  155197007  ENSG00000155959.10    0.120559    0.249427   \n",
       "15024  chrX  155264588  155264589   ENSG00000155961.4    0.807824    1.062786   \n",
       "15025  chrX  155334656  155334657  ENSG00000155962.12   -0.057000   -0.107816   \n",
       "15026  chrX  155669943  155669944  ENSG00000185973.10    0.451658    0.315311   \n",
       "15027  chrX  155767811  155767812  ENSG00000168939.11    0.465723    0.107816   \n",
       "\n",
       "       GTEX-11DZ1  GTEX-11GSP  GTEX-11NUK  GTEX-11OF3  ...  GTEX-ZAB4  \\\n",
       "0       -1.062786   -0.643034    0.382595   -0.423792  ...  -0.861634   \n",
       "1        0.597003   -0.107816    0.031655   -0.082381  ...   0.082381   \n",
       "2        0.315311   -0.006330   -1.207414   -0.031655  ...  -0.107816   \n",
       "3        1.529007   -0.451658   -0.597003    0.706628  ...   0.690469   \n",
       "4        0.369009    0.756257   -2.428737    1.207414  ...  -0.508488   \n",
       "...           ...         ...         ...         ...  ...        ...   \n",
       "15023   -1.062786   -1.132140    1.234084   -1.156505  ...   1.207414   \n",
       "15024    0.937532   -0.315311    1.452133    0.057000  ...   1.234084   \n",
       "15025    0.018991   -0.210428   -1.181577   -0.275635  ...  -1.616156   \n",
       "15026   -0.315311   -0.146105    1.181577   -0.184609  ...  -1.062786   \n",
       "15027    0.790402   -0.158913    1.019257    0.756257  ...  -0.552205   \n",
       "\n",
       "       GTEX-ZAK1  GTEX-ZDXO  GTEX-ZE7O  GTEX-ZF28  GTEX-ZUA1  GTEX-ZV68  \\\n",
       "0      -1.840761  -0.120559   0.288809   0.342037  -0.396252   0.355490   \n",
       "1       0.146105  -0.627544   0.552205   1.416717   0.382595   0.369009   \n",
       "2      -0.249427   0.018991   0.508488   0.262509   0.184609   0.612202   \n",
       "3      -0.465723   0.998201   0.790402   1.717660   0.133321   0.288809   \n",
       "4       1.234084  -0.031655   0.977579   0.825494   0.522948   0.567010   \n",
       "...          ...        ...        ...        ...        ...        ...   \n",
       "15023  -0.210428  -0.739512  -0.508488  -0.825494   0.275635  -0.018991   \n",
       "15024  -0.658680  -0.396252  -0.581942  -1.207414  -0.031655  -1.416717   \n",
       "15025  -0.957363  -1.085325  -0.537519   1.108429  -0.158913   0.031655   \n",
       "15026   0.581942  -0.197502  -1.914506  -0.937532  -0.107816   0.275635   \n",
       "15027  -1.452133  -0.275635  -0.302034   0.057000   0.522948  -0.465723   \n",
       "\n",
       "       GTEX-ZVT3  GTEX-ZVZQ  GTEX-ZXG5  \n",
       "0      -0.044324  -0.807824   0.031655  \n",
       "1       0.465723  -0.508488  -0.069685  \n",
       "2       0.494133   1.181577   0.369009  \n",
       "3       0.423792   0.722972   0.756257  \n",
       "4       0.658680   0.146105   0.612202  \n",
       "...          ...        ...        ...  \n",
       "15023  -0.756257   0.739512  -0.451658  \n",
       "15024  -1.840761   0.627544  -0.236388  \n",
       "15025   0.249427  -0.522948  -0.184609  \n",
       "15026   0.044324   1.664765  -0.627544  \n",
       "15027  -0.095091   1.914506   0.328645  \n",
       "\n",
       "[15028 rows x 202 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "region_pick = ['Amygdala', 'Anterior_cingulate_cortex_BA24', 'Caudate_basal_ganglia', \n",
    "               'Cerebellar_Hemisphere', 'Frontal_Cortex_BA9', 'Hippocampus', 'Hypothalamus', \n",
    "               'Nucleus_accumbens_basal_ganglia', 'Putamen_basal_ganglia', 'Substantia_nigra']\n",
    "\n",
    "pred_region = region_pick[0]\n",
    "save_dir = '/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/02162024/eqtl_analysis/new_bedgz_file/'\n",
    "filename = f'{pred_region}_newnormed.expression.bed.gz'\n",
    "mat1 = pd.read_csv(save_dir+filename, sep='\\t', compression='gzip')\n",
    "# filename = f'{pred_region}_newnormed.expression.sorted.bed.gz'\n",
    "# mat2 = pd.read_csv(save_dir+filename, sep='\\t', compression='gzip')\n",
    "mat1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7c5369fc-3eff-4d3e-a475-750fa143122b",
   "metadata": {},
   "source": [
    "### Generate a mini-batch bed.gz file for testing!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8814578a-da94-4692-8bcb-227fdc661111",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "save_dir = '/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/02162024/eqtl_analysis/GTEx_Analysis_v8_eQTL_expression_matrices/'\n",
    "filename = 'Brain_Amygdala.v8.normalized_expression.bed.gz'\n",
    "gtex_df = pd.read_csv(save_dir+filename, sep='\\t', compression='gzip')\n",
    "mini_df = gtex_df.loc[:100,]\n",
    "# save the df\n",
    "save_dir = '/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/02162024/eqtl_analysis/Test_GTEx_fastQTL_results/'\n",
    "mini_df_filename='Brain_Amygdala.v8.mini.normalized_expression.bed.gz'\n",
    "mini_df.to_csv(save_dir+mini_df_filename, sep='\\t', index=False, header=True, compression='gzip')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "66e304cf-1cb6-4eb0-af7c-ce04aa1c7057",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GTEX-11NUK</th>\n",
       "      <th>GTEX-11ONC</th>\n",
       "      <th>GTEX-11PRG</th>\n",
       "      <th>GTEX-11TTK</th>\n",
       "      <th>GTEX-11ZU8</th>\n",
       "      <th>GTEX-11ZVC</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-WZTO</th>\n",
       "      <th>GTEX-X4EP</th>\n",
       "      <th>GTEX-X4XX</th>\n",
       "      <th>GTEX-X585</th>\n",
       "      <th>GTEX-XLM4</th>\n",
       "      <th>GTEX-Y8DK</th>\n",
       "      <th>GTEX-YFC4</th>\n",
       "      <th>GTEX-ZAB4</th>\n",
       "      <th>GTEX-ZAK1</th>\n",
       "      <th>GTEX-ZVZQ</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29552</td>\n",
       "      <td>29553</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-1.159742</td>\n",
       "      <td>-0.736316</td>\n",
       "      <td>-0.174416</td>\n",
       "      <td>0.524401</td>\n",
       "      <td>-1.238894</td>\n",
       "      <td>-0.686643</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.569185</td>\n",
       "      <td>0.194028</td>\n",
       "      <td>1.281552</td>\n",
       "      <td>1.020076</td>\n",
       "      <td>0.313572</td>\n",
       "      <td>1.122762</td>\n",
       "      <td>0.480644</td>\n",
       "      <td>1.159742</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>-0.333891</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>62947</td>\n",
       "      <td>62948</td>\n",
       "      <td>ENSG00000240361.1</td>\n",
       "      <td>0.135385</td>\n",
       "      <td>-0.546656</td>\n",
       "      <td>0.459112</td>\n",
       "      <td>0.761843</td>\n",
       "      <td>1.122762</td>\n",
       "      <td>0.736316</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>-0.038573</td>\n",
       "      <td>-0.761843</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>-0.077204</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>-0.615141</td>\n",
       "      <td>0.057878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69090</td>\n",
       "      <td>69091</td>\n",
       "      <td>ENSG00000186092.4</td>\n",
       "      <td>0.233485</td>\n",
       "      <td>2.160044</td>\n",
       "      <td>-1.159742</td>\n",
       "      <td>0.253347</td>\n",
       "      <td>0.761843</td>\n",
       "      <td>0.897915</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.057878</td>\n",
       "      <td>-1.608653</td>\n",
       "      <td>-0.213715</td>\n",
       "      <td>-1.608653</td>\n",
       "      <td>-0.233485</td>\n",
       "      <td>-1.608653</td>\n",
       "      <td>0.480644</td>\n",
       "      <td>-0.686643</td>\n",
       "      <td>-1.608653</td>\n",
       "      <td>-0.592007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135894</td>\n",
       "      <td>135895</td>\n",
       "      <td>ENSG00000268903.1</td>\n",
       "      <td>-0.213715</td>\n",
       "      <td>0.019283</td>\n",
       "      <td>-1.238894</td>\n",
       "      <td>0.957209</td>\n",
       "      <td>-1.541986</td>\n",
       "      <td>-1.374680</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.354350</td>\n",
       "      <td>-0.437790</td>\n",
       "      <td>0.927155</td>\n",
       "      <td>1.683348</td>\n",
       "      <td>0.736316</td>\n",
       "      <td>0.194028</td>\n",
       "      <td>0.213715</td>\n",
       "      <td>0.135385</td>\n",
       "      <td>-0.416666</td>\n",
       "      <td>-0.592007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>137964</td>\n",
       "      <td>137965</td>\n",
       "      <td>ENSG00000269981.1</td>\n",
       "      <td>-0.115950</td>\n",
       "      <td>-0.077204</td>\n",
       "      <td>-0.897915</td>\n",
       "      <td>0.437790</td>\n",
       "      <td>-0.395725</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.174416</td>\n",
       "      <td>0.841621</td>\n",
       "      <td>0.354350</td>\n",
       "      <td>2.160044</td>\n",
       "      <td>0.135385</td>\n",
       "      <td>-0.502402</td>\n",
       "      <td>0.194028</td>\n",
       "      <td>-0.761843</td>\n",
       "      <td>-0.333891</td>\n",
       "      <td>-0.293381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2015529</td>\n",
       "      <td>2015530</td>\n",
       "      <td>ENSG00000233542.1</td>\n",
       "      <td>0.988155</td>\n",
       "      <td>-0.233485</td>\n",
       "      <td>-0.019283</td>\n",
       "      <td>-1.238894</td>\n",
       "      <td>-1.869607</td>\n",
       "      <td>-1.198380</td>\n",
       "      <td>...</td>\n",
       "      <td>2.160044</td>\n",
       "      <td>-1.993984</td>\n",
       "      <td>0.662435</td>\n",
       "      <td>0.374957</td>\n",
       "      <td>0.927155</td>\n",
       "      <td>-0.416666</td>\n",
       "      <td>0.592007</td>\n",
       "      <td>-0.213715</td>\n",
       "      <td>0.077204</td>\n",
       "      <td>0.897915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2019297</td>\n",
       "      <td>2019298</td>\n",
       "      <td>ENSG00000187730.8</td>\n",
       "      <td>1.159742</td>\n",
       "      <td>-0.313572</td>\n",
       "      <td>1.198380</td>\n",
       "      <td>-0.869424</td>\n",
       "      <td>-1.426077</td>\n",
       "      <td>-1.869607</td>\n",
       "      <td>...</td>\n",
       "      <td>2.423196</td>\n",
       "      <td>-2.423196</td>\n",
       "      <td>-0.711260</td>\n",
       "      <td>-0.615141</td>\n",
       "      <td>0.841621</td>\n",
       "      <td>0.115950</td>\n",
       "      <td>0.814455</td>\n",
       "      <td>-0.437790</td>\n",
       "      <td>-0.662435</td>\n",
       "      <td>0.057878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2050469</td>\n",
       "      <td>2050470</td>\n",
       "      <td>ENSG00000067606.16</td>\n",
       "      <td>1.159742</td>\n",
       "      <td>-0.480644</td>\n",
       "      <td>0.524401</td>\n",
       "      <td>-1.481544</td>\n",
       "      <td>-1.281552</td>\n",
       "      <td>-1.869607</td>\n",
       "      <td>...</td>\n",
       "      <td>1.768825</td>\n",
       "      <td>-1.053073</td>\n",
       "      <td>-0.135385</td>\n",
       "      <td>0.135385</td>\n",
       "      <td>0.374957</td>\n",
       "      <td>-0.502402</td>\n",
       "      <td>0.736316</td>\n",
       "      <td>1.326678</td>\n",
       "      <td>-0.293381</td>\n",
       "      <td>0.077204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2184388</td>\n",
       "      <td>2184389</td>\n",
       "      <td>ENSG00000182873.5</td>\n",
       "      <td>-1.374680</td>\n",
       "      <td>0.135385</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>-0.354350</td>\n",
       "      <td>-1.281552</td>\n",
       "      <td>-1.122762</td>\n",
       "      <td>...</td>\n",
       "      <td>0.154871</td>\n",
       "      <td>-0.957209</td>\n",
       "      <td>2.423196</td>\n",
       "      <td>1.768825</td>\n",
       "      <td>0.869424</td>\n",
       "      <td>1.608653</td>\n",
       "      <td>0.416666</td>\n",
       "      <td>1.683348</td>\n",
       "      <td>-0.253347</td>\n",
       "      <td>-0.988155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2212719</td>\n",
       "      <td>2212720</td>\n",
       "      <td>ENSG00000162585.16</td>\n",
       "      <td>-0.416666</td>\n",
       "      <td>-0.057878</td>\n",
       "      <td>-0.154871</td>\n",
       "      <td>0.416666</td>\n",
       "      <td>1.869607</td>\n",
       "      <td>0.057878</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.957209</td>\n",
       "      <td>-0.988155</td>\n",
       "      <td>0.686643</td>\n",
       "      <td>0.019283</td>\n",
       "      <td>-0.869424</td>\n",
       "      <td>1.541986</td>\n",
       "      <td>-1.608653</td>\n",
       "      <td>0.154871</td>\n",
       "      <td>0.194028</td>\n",
       "      <td>-0.395725</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>101 rows × 133 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #chr    start      end             gene_id  GTEX-11NUK  GTEX-11ONC  \\\n",
       "0    chr1    29552    29553   ENSG00000227232.5   -1.159742   -0.736316   \n",
       "1    chr1    62947    62948   ENSG00000240361.1    0.135385   -0.546656   \n",
       "2    chr1    69090    69091   ENSG00000186092.4    0.233485    2.160044   \n",
       "3    chr1   135894   135895   ENSG00000268903.1   -0.213715    0.019283   \n",
       "4    chr1   137964   137965   ENSG00000269981.1   -0.115950   -0.077204   \n",
       "..    ...      ...      ...                 ...         ...         ...   \n",
       "96   chr1  2015529  2015530   ENSG00000233542.1    0.988155   -0.233485   \n",
       "97   chr1  2019297  2019298   ENSG00000187730.8    1.159742   -0.313572   \n",
       "98   chr1  2050469  2050470  ENSG00000067606.16    1.159742   -0.480644   \n",
       "99   chr1  2184388  2184389   ENSG00000182873.5   -1.374680    0.135385   \n",
       "100  chr1  2212719  2212720  ENSG00000162585.16   -0.416666   -0.057878   \n",
       "\n",
       "     GTEX-11PRG  GTEX-11TTK  GTEX-11ZU8  GTEX-11ZVC  ...  GTEX-WZTO  \\\n",
       "0     -0.174416    0.524401   -1.238894   -0.686643  ...  -0.569185   \n",
       "1      0.459112    0.761843    1.122762    0.736316  ...  -1.326678   \n",
       "2     -1.159742    0.253347    0.761843    0.897915  ...  -0.057878   \n",
       "3     -1.238894    0.957209   -1.541986   -1.374680  ...  -0.354350   \n",
       "4     -0.897915    0.437790   -0.395725   -1.326678  ...  -0.174416   \n",
       "..          ...         ...         ...         ...  ...        ...   \n",
       "96    -0.019283   -1.238894   -1.869607   -1.198380  ...   2.160044   \n",
       "97     1.198380   -0.869424   -1.426077   -1.869607  ...   2.423196   \n",
       "98     0.524401   -1.481544   -1.281552   -1.869607  ...   1.768825   \n",
       "99    -1.326678   -0.354350   -1.281552   -1.122762  ...   0.154871   \n",
       "100   -0.154871    0.416666    1.869607    0.057878  ...  -0.957209   \n",
       "\n",
       "     GTEX-X4EP  GTEX-X4XX  GTEX-X585  GTEX-XLM4  GTEX-Y8DK  GTEX-YFC4  \\\n",
       "0     0.194028   1.281552   1.020076   0.313572   1.122762   0.480644   \n",
       "1    -1.326678  -0.038573  -0.761843  -1.326678  -1.326678  -0.077204   \n",
       "2    -1.608653  -0.213715  -1.608653  -0.233485  -1.608653   0.480644   \n",
       "3    -0.437790   0.927155   1.683348   0.736316   0.194028   0.213715   \n",
       "4     0.841621   0.354350   2.160044   0.135385  -0.502402   0.194028   \n",
       "..         ...        ...        ...        ...        ...        ...   \n",
       "96   -1.993984   0.662435   0.374957   0.927155  -0.416666   0.592007   \n",
       "97   -2.423196  -0.711260  -0.615141   0.841621   0.115950   0.814455   \n",
       "98   -1.053073  -0.135385   0.135385   0.374957  -0.502402   0.736316   \n",
       "99   -0.957209   2.423196   1.768825   0.869424   1.608653   0.416666   \n",
       "100  -0.988155   0.686643   0.019283  -0.869424   1.541986  -1.608653   \n",
       "\n",
       "     GTEX-ZAB4  GTEX-ZAK1  GTEX-ZVZQ  \n",
       "0     1.159742  -1.326678  -0.333891  \n",
       "1    -1.326678  -0.615141   0.057878  \n",
       "2    -0.686643  -1.608653  -0.592007  \n",
       "3     0.135385  -0.416666  -0.592007  \n",
       "4    -0.761843  -0.333891  -0.293381  \n",
       "..         ...        ...        ...  \n",
       "96   -0.213715   0.077204   0.897915  \n",
       "97   -0.437790  -0.662435   0.057878  \n",
       "98    1.326678  -0.293381   0.077204  \n",
       "99    1.683348  -0.253347  -0.988155  \n",
       "100   0.154871   0.194028  -0.395725  \n",
       "\n",
       "[101 rows x 133 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7662e7a0-e895-4687-a930-e4baefa74f4e",
   "metadata": {},
   "source": [
    "### compare our bed.gz file with GTEx bed.gz file\n",
    "#### to see how often are the same gene-snp pairs in both files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "acc9661d-0d99-408c-8f63-5d0b8da13500",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "region_pick = ['Amygdala', 'Anterior_cingulate_cortex_BA24', 'Caudate_basal_ganglia', \n",
    "               'Cerebellar_Hemisphere', 'Frontal_Cortex_BA9', 'Hippocampus', 'Hypothalamus', \n",
    "               'Nucleus_accumbens_basal_ganglia', 'Putamen_basal_ganglia', 'Substantia_nigra']\n",
    "\n",
    "pred_region = region_pick[0]\n",
    "save_dir = '/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/02162024/eqtl_analysis/new_bedgz_file/'\n",
    "filename = f'{pred_region}_newnormed.expression.bed.gz'\n",
    "our_bed = pd.read_csv(save_dir+filename, sep='\\t', compression='gzip')\n",
    "\n",
    "save_dir = '/project/pi_rachel_melamed_uml_edu/Jianfeng/Allen/src/Pytorch/02162024/eqtl_analysis/GTEx_Analysis_v8_eQTL_expression_matrices/'\n",
    "filename = 'Brain_Amygdala.v8.normalized_expression.bed.gz'\n",
    "GTEx_bed = pd.read_csv(save_dir+filename, sep='\\t', compression='gzip')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "11c8b7f0-43be-4a7c-9ce5-fdd653385286",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GTEX-1192X</th>\n",
       "      <th>GTEX-11DYG</th>\n",
       "      <th>GTEX-11DZ1</th>\n",
       "      <th>GTEX-11GSP</th>\n",
       "      <th>GTEX-11NUK</th>\n",
       "      <th>GTEX-11OF3</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-ZAB4</th>\n",
       "      <th>GTEX-ZAK1</th>\n",
       "      <th>GTEX-ZDXO</th>\n",
       "      <th>GTEX-ZE7O</th>\n",
       "      <th>GTEX-ZF28</th>\n",
       "      <th>GTEX-ZUA1</th>\n",
       "      <th>GTEX-ZV68</th>\n",
       "      <th>GTEX-ZVT3</th>\n",
       "      <th>GTEX-ZVZQ</th>\n",
       "      <th>GTEX-ZXG5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69090</td>\n",
       "      <td>69091</td>\n",
       "      <td>ENSG00000186092.4</td>\n",
       "      <td>-0.369009</td>\n",
       "      <td>-0.898936</td>\n",
       "      <td>-1.062786</td>\n",
       "      <td>-0.643034</td>\n",
       "      <td>0.382595</td>\n",
       "      <td>-0.423792</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.861634</td>\n",
       "      <td>-1.840761</td>\n",
       "      <td>-0.120559</td>\n",
       "      <td>0.288809</td>\n",
       "      <td>0.342037</td>\n",
       "      <td>-0.396252</td>\n",
       "      <td>0.355490</td>\n",
       "      <td>-0.044324</td>\n",
       "      <td>-0.807824</td>\n",
       "      <td>0.031655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>827521</td>\n",
       "      <td>827522</td>\n",
       "      <td>ENSG00000225880.5</td>\n",
       "      <td>-0.249427</td>\n",
       "      <td>-1.350775</td>\n",
       "      <td>0.597003</td>\n",
       "      <td>-0.107816</td>\n",
       "      <td>0.031655</td>\n",
       "      <td>-0.082381</td>\n",
       "      <td>...</td>\n",
       "      <td>0.082381</td>\n",
       "      <td>0.146105</td>\n",
       "      <td>-0.627544</td>\n",
       "      <td>0.552205</td>\n",
       "      <td>1.416717</td>\n",
       "      <td>0.382595</td>\n",
       "      <td>0.369009</td>\n",
       "      <td>0.465723</td>\n",
       "      <td>-0.508488</td>\n",
       "      <td>-0.069685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>876753</td>\n",
       "      <td>876754</td>\n",
       "      <td>ENSG00000230368.2</td>\n",
       "      <td>-0.807824</td>\n",
       "      <td>-1.156505</td>\n",
       "      <td>0.315311</td>\n",
       "      <td>-0.006330</td>\n",
       "      <td>-1.207414</td>\n",
       "      <td>-0.031655</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.107816</td>\n",
       "      <td>-0.249427</td>\n",
       "      <td>0.018991</td>\n",
       "      <td>0.508488</td>\n",
       "      <td>0.262509</td>\n",
       "      <td>0.184609</td>\n",
       "      <td>0.612202</td>\n",
       "      <td>0.494133</td>\n",
       "      <td>1.181577</td>\n",
       "      <td>0.369009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>911434</td>\n",
       "      <td>911435</td>\n",
       "      <td>ENSG00000230699.2</td>\n",
       "      <td>-0.382595</td>\n",
       "      <td>0.918063</td>\n",
       "      <td>1.529007</td>\n",
       "      <td>-0.451658</td>\n",
       "      <td>-0.597003</td>\n",
       "      <td>0.706628</td>\n",
       "      <td>...</td>\n",
       "      <td>0.690469</td>\n",
       "      <td>-0.465723</td>\n",
       "      <td>0.998201</td>\n",
       "      <td>0.790402</td>\n",
       "      <td>1.717660</td>\n",
       "      <td>0.133321</td>\n",
       "      <td>0.288809</td>\n",
       "      <td>0.423792</td>\n",
       "      <td>0.722972</td>\n",
       "      <td>0.756257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923927</td>\n",
       "      <td>923928</td>\n",
       "      <td>ENSG00000187634.11</td>\n",
       "      <td>0.479879</td>\n",
       "      <td>0.581942</td>\n",
       "      <td>0.369009</td>\n",
       "      <td>0.756257</td>\n",
       "      <td>-2.428737</td>\n",
       "      <td>1.207414</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.508488</td>\n",
       "      <td>1.234084</td>\n",
       "      <td>-0.031655</td>\n",
       "      <td>0.977579</td>\n",
       "      <td>0.825494</td>\n",
       "      <td>0.522948</td>\n",
       "      <td>0.567010</td>\n",
       "      <td>0.658680</td>\n",
       "      <td>0.146105</td>\n",
       "      <td>0.612202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15023</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155197006</td>\n",
       "      <td>155197007</td>\n",
       "      <td>ENSG00000155959.10</td>\n",
       "      <td>0.120559</td>\n",
       "      <td>0.249427</td>\n",
       "      <td>-1.062786</td>\n",
       "      <td>-1.132140</td>\n",
       "      <td>1.234084</td>\n",
       "      <td>-1.156505</td>\n",
       "      <td>...</td>\n",
       "      <td>1.207414</td>\n",
       "      <td>-0.210428</td>\n",
       "      <td>-0.739512</td>\n",
       "      <td>-0.508488</td>\n",
       "      <td>-0.825494</td>\n",
       "      <td>0.275635</td>\n",
       "      <td>-0.018991</td>\n",
       "      <td>-0.756257</td>\n",
       "      <td>0.739512</td>\n",
       "      <td>-0.451658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15024</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155264588</td>\n",
       "      <td>155264589</td>\n",
       "      <td>ENSG00000155961.4</td>\n",
       "      <td>0.807824</td>\n",
       "      <td>1.062786</td>\n",
       "      <td>0.937532</td>\n",
       "      <td>-0.315311</td>\n",
       "      <td>1.452133</td>\n",
       "      <td>0.057000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.234084</td>\n",
       "      <td>-0.658680</td>\n",
       "      <td>-0.396252</td>\n",
       "      <td>-0.581942</td>\n",
       "      <td>-1.207414</td>\n",
       "      <td>-0.031655</td>\n",
       "      <td>-1.416717</td>\n",
       "      <td>-1.840761</td>\n",
       "      <td>0.627544</td>\n",
       "      <td>-0.236388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15025</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155334656</td>\n",
       "      <td>155334657</td>\n",
       "      <td>ENSG00000155962.12</td>\n",
       "      <td>-0.057000</td>\n",
       "      <td>-0.107816</td>\n",
       "      <td>0.018991</td>\n",
       "      <td>-0.210428</td>\n",
       "      <td>-1.181577</td>\n",
       "      <td>-0.275635</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.616156</td>\n",
       "      <td>-0.957363</td>\n",
       "      <td>-1.085325</td>\n",
       "      <td>-0.537519</td>\n",
       "      <td>1.108429</td>\n",
       "      <td>-0.158913</td>\n",
       "      <td>0.031655</td>\n",
       "      <td>0.249427</td>\n",
       "      <td>-0.522948</td>\n",
       "      <td>-0.184609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15026</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155669943</td>\n",
       "      <td>155669944</td>\n",
       "      <td>ENSG00000185973.10</td>\n",
       "      <td>0.451658</td>\n",
       "      <td>0.315311</td>\n",
       "      <td>-0.315311</td>\n",
       "      <td>-0.146105</td>\n",
       "      <td>1.181577</td>\n",
       "      <td>-0.184609</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.062786</td>\n",
       "      <td>0.581942</td>\n",
       "      <td>-0.197502</td>\n",
       "      <td>-1.914506</td>\n",
       "      <td>-0.937532</td>\n",
       "      <td>-0.107816</td>\n",
       "      <td>0.275635</td>\n",
       "      <td>0.044324</td>\n",
       "      <td>1.664765</td>\n",
       "      <td>-0.627544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15027</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155767811</td>\n",
       "      <td>155767812</td>\n",
       "      <td>ENSG00000168939.11</td>\n",
       "      <td>0.465723</td>\n",
       "      <td>0.107816</td>\n",
       "      <td>0.790402</td>\n",
       "      <td>-0.158913</td>\n",
       "      <td>1.019257</td>\n",
       "      <td>0.756257</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.552205</td>\n",
       "      <td>-1.452133</td>\n",
       "      <td>-0.275635</td>\n",
       "      <td>-0.302034</td>\n",
       "      <td>0.057000</td>\n",
       "      <td>0.522948</td>\n",
       "      <td>-0.465723</td>\n",
       "      <td>-0.095091</td>\n",
       "      <td>1.914506</td>\n",
       "      <td>0.328645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15028 rows × 202 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #chr      start        end             gene_id  GTEX-1192X  GTEX-11DYG  \\\n",
       "0      chr1      69090      69091   ENSG00000186092.4   -0.369009   -0.898936   \n",
       "1      chr1     827521     827522   ENSG00000225880.5   -0.249427   -1.350775   \n",
       "2      chr1     876753     876754   ENSG00000230368.2   -0.807824   -1.156505   \n",
       "3      chr1     911434     911435   ENSG00000230699.2   -0.382595    0.918063   \n",
       "4      chr1     923927     923928  ENSG00000187634.11    0.479879    0.581942   \n",
       "...     ...        ...        ...                 ...         ...         ...   \n",
       "15023  chrX  155197006  155197007  ENSG00000155959.10    0.120559    0.249427   \n",
       "15024  chrX  155264588  155264589   ENSG00000155961.4    0.807824    1.062786   \n",
       "15025  chrX  155334656  155334657  ENSG00000155962.12   -0.057000   -0.107816   \n",
       "15026  chrX  155669943  155669944  ENSG00000185973.10    0.451658    0.315311   \n",
       "15027  chrX  155767811  155767812  ENSG00000168939.11    0.465723    0.107816   \n",
       "\n",
       "       GTEX-11DZ1  GTEX-11GSP  GTEX-11NUK  GTEX-11OF3  ...  GTEX-ZAB4  \\\n",
       "0       -1.062786   -0.643034    0.382595   -0.423792  ...  -0.861634   \n",
       "1        0.597003   -0.107816    0.031655   -0.082381  ...   0.082381   \n",
       "2        0.315311   -0.006330   -1.207414   -0.031655  ...  -0.107816   \n",
       "3        1.529007   -0.451658   -0.597003    0.706628  ...   0.690469   \n",
       "4        0.369009    0.756257   -2.428737    1.207414  ...  -0.508488   \n",
       "...           ...         ...         ...         ...  ...        ...   \n",
       "15023   -1.062786   -1.132140    1.234084   -1.156505  ...   1.207414   \n",
       "15024    0.937532   -0.315311    1.452133    0.057000  ...   1.234084   \n",
       "15025    0.018991   -0.210428   -1.181577   -0.275635  ...  -1.616156   \n",
       "15026   -0.315311   -0.146105    1.181577   -0.184609  ...  -1.062786   \n",
       "15027    0.790402   -0.158913    1.019257    0.756257  ...  -0.552205   \n",
       "\n",
       "       GTEX-ZAK1  GTEX-ZDXO  GTEX-ZE7O  GTEX-ZF28  GTEX-ZUA1  GTEX-ZV68  \\\n",
       "0      -1.840761  -0.120559   0.288809   0.342037  -0.396252   0.355490   \n",
       "1       0.146105  -0.627544   0.552205   1.416717   0.382595   0.369009   \n",
       "2      -0.249427   0.018991   0.508488   0.262509   0.184609   0.612202   \n",
       "3      -0.465723   0.998201   0.790402   1.717660   0.133321   0.288809   \n",
       "4       1.234084  -0.031655   0.977579   0.825494   0.522948   0.567010   \n",
       "...          ...        ...        ...        ...        ...        ...   \n",
       "15023  -0.210428  -0.739512  -0.508488  -0.825494   0.275635  -0.018991   \n",
       "15024  -0.658680  -0.396252  -0.581942  -1.207414  -0.031655  -1.416717   \n",
       "15025  -0.957363  -1.085325  -0.537519   1.108429  -0.158913   0.031655   \n",
       "15026   0.581942  -0.197502  -1.914506  -0.937532  -0.107816   0.275635   \n",
       "15027  -1.452133  -0.275635  -0.302034   0.057000   0.522948  -0.465723   \n",
       "\n",
       "       GTEX-ZVT3  GTEX-ZVZQ  GTEX-ZXG5  \n",
       "0      -0.044324  -0.807824   0.031655  \n",
       "1       0.465723  -0.508488  -0.069685  \n",
       "2       0.494133   1.181577   0.369009  \n",
       "3       0.423792   0.722972   0.756257  \n",
       "4       0.658680   0.146105   0.612202  \n",
       "...          ...        ...        ...  \n",
       "15023  -0.756257   0.739512  -0.451658  \n",
       "15024  -1.840761   0.627544  -0.236388  \n",
       "15025   0.249427  -0.522948  -0.184609  \n",
       "15026   0.044324   1.664765  -0.627544  \n",
       "15027  -0.095091   1.914506   0.328645  \n",
       "\n",
       "[15028 rows x 202 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "our_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bada576c-7391-42fd-bf00-fb9f6997c774",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GTEX-11NUK</th>\n",
       "      <th>GTEX-11ONC</th>\n",
       "      <th>GTEX-11PRG</th>\n",
       "      <th>GTEX-11TTK</th>\n",
       "      <th>GTEX-11ZU8</th>\n",
       "      <th>GTEX-11ZVC</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-WZTO</th>\n",
       "      <th>GTEX-X4EP</th>\n",
       "      <th>GTEX-X4XX</th>\n",
       "      <th>GTEX-X585</th>\n",
       "      <th>GTEX-XLM4</th>\n",
       "      <th>GTEX-Y8DK</th>\n",
       "      <th>GTEX-YFC4</th>\n",
       "      <th>GTEX-ZAB4</th>\n",
       "      <th>GTEX-ZAK1</th>\n",
       "      <th>GTEX-ZVZQ</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29552</td>\n",
       "      <td>29553</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>-1.159742</td>\n",
       "      <td>-0.736316</td>\n",
       "      <td>-0.174416</td>\n",
       "      <td>0.524401</td>\n",
       "      <td>-1.238894</td>\n",
       "      <td>-0.686643</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.569185</td>\n",
       "      <td>0.194028</td>\n",
       "      <td>1.281552</td>\n",
       "      <td>1.020076</td>\n",
       "      <td>0.313572</td>\n",
       "      <td>1.122762</td>\n",
       "      <td>0.480644</td>\n",
       "      <td>1.159742</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>-0.333891</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>62947</td>\n",
       "      <td>62948</td>\n",
       "      <td>ENSG00000240361.1</td>\n",
       "      <td>0.135385</td>\n",
       "      <td>-0.546656</td>\n",
       "      <td>0.459112</td>\n",
       "      <td>0.761843</td>\n",
       "      <td>1.122762</td>\n",
       "      <td>0.736316</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>-0.038573</td>\n",
       "      <td>-0.761843</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>-0.077204</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>-0.615141</td>\n",
       "      <td>0.057878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69090</td>\n",
       "      <td>69091</td>\n",
       "      <td>ENSG00000186092.4</td>\n",
       "      <td>0.233485</td>\n",
       "      <td>2.160044</td>\n",
       "      <td>-1.159742</td>\n",
       "      <td>0.253347</td>\n",
       "      <td>0.761843</td>\n",
       "      <td>0.897915</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.057878</td>\n",
       "      <td>-1.608653</td>\n",
       "      <td>-0.213715</td>\n",
       "      <td>-1.608653</td>\n",
       "      <td>-0.233485</td>\n",
       "      <td>-1.608653</td>\n",
       "      <td>0.480644</td>\n",
       "      <td>-0.686643</td>\n",
       "      <td>-1.608653</td>\n",
       "      <td>-0.592007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135894</td>\n",
       "      <td>135895</td>\n",
       "      <td>ENSG00000268903.1</td>\n",
       "      <td>-0.213715</td>\n",
       "      <td>0.019283</td>\n",
       "      <td>-1.238894</td>\n",
       "      <td>0.957209</td>\n",
       "      <td>-1.541986</td>\n",
       "      <td>-1.374680</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.354350</td>\n",
       "      <td>-0.437790</td>\n",
       "      <td>0.927155</td>\n",
       "      <td>1.683348</td>\n",
       "      <td>0.736316</td>\n",
       "      <td>0.194028</td>\n",
       "      <td>0.213715</td>\n",
       "      <td>0.135385</td>\n",
       "      <td>-0.416666</td>\n",
       "      <td>-0.592007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>137964</td>\n",
       "      <td>137965</td>\n",
       "      <td>ENSG00000269981.1</td>\n",
       "      <td>-0.115950</td>\n",
       "      <td>-0.077204</td>\n",
       "      <td>-0.897915</td>\n",
       "      <td>0.437790</td>\n",
       "      <td>-0.395725</td>\n",
       "      <td>-1.326678</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.174416</td>\n",
       "      <td>0.841621</td>\n",
       "      <td>0.354350</td>\n",
       "      <td>2.160044</td>\n",
       "      <td>0.135385</td>\n",
       "      <td>-0.502402</td>\n",
       "      <td>0.194028</td>\n",
       "      <td>-0.761843</td>\n",
       "      <td>-0.333891</td>\n",
       "      <td>-0.293381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24064</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155466539</td>\n",
       "      <td>155466540</td>\n",
       "      <td>ENSG00000224533.4</td>\n",
       "      <td>-0.736316</td>\n",
       "      <td>-1.218387</td>\n",
       "      <td>-0.524401</td>\n",
       "      <td>-1.218387</td>\n",
       "      <td>1.087257</td>\n",
       "      <td>1.159742</td>\n",
       "      <td>...</td>\n",
       "      <td>0.841621</td>\n",
       "      <td>0.313572</td>\n",
       "      <td>-1.218387</td>\n",
       "      <td>0.194028</td>\n",
       "      <td>-1.218387</td>\n",
       "      <td>-1.218387</td>\n",
       "      <td>0.019283</td>\n",
       "      <td>-1.218387</td>\n",
       "      <td>-1.218387</td>\n",
       "      <td>0.686643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24065</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155669943</td>\n",
       "      <td>155669944</td>\n",
       "      <td>ENSG00000185973.10</td>\n",
       "      <td>1.020076</td>\n",
       "      <td>-1.198380</td>\n",
       "      <td>0.869424</td>\n",
       "      <td>2.423196</td>\n",
       "      <td>-0.019283</td>\n",
       "      <td>0.416666</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.333891</td>\n",
       "      <td>0.615141</td>\n",
       "      <td>-1.768825</td>\n",
       "      <td>-1.281552</td>\n",
       "      <td>0.437790</td>\n",
       "      <td>0.115950</td>\n",
       "      <td>-0.502402</td>\n",
       "      <td>-1.159742</td>\n",
       "      <td>0.154871</td>\n",
       "      <td>1.426077</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24066</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155767811</td>\n",
       "      <td>155767812</td>\n",
       "      <td>ENSG00000168939.11</td>\n",
       "      <td>1.053073</td>\n",
       "      <td>-0.761843</td>\n",
       "      <td>1.238894</td>\n",
       "      <td>-1.374680</td>\n",
       "      <td>-0.174416</td>\n",
       "      <td>0.638610</td>\n",
       "      <td>...</td>\n",
       "      <td>1.541986</td>\n",
       "      <td>-0.927155</td>\n",
       "      <td>0.761843</td>\n",
       "      <td>-1.159742</td>\n",
       "      <td>-0.213715</td>\n",
       "      <td>0.897915</td>\n",
       "      <td>0.711260</td>\n",
       "      <td>-0.253347</td>\n",
       "      <td>-1.281552</td>\n",
       "      <td>1.768825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24067</th>\n",
       "      <td>chrX</td>\n",
       "      <td>155881292</td>\n",
       "      <td>155881293</td>\n",
       "      <td>ENSG00000124333.15</td>\n",
       "      <td>0.736316</td>\n",
       "      <td>-0.135385</td>\n",
       "      <td>0.233485</td>\n",
       "      <td>0.459112</td>\n",
       "      <td>1.541986</td>\n",
       "      <td>0.374957</td>\n",
       "      <td>...</td>\n",
       "      <td>0.313572</td>\n",
       "      <td>-1.541986</td>\n",
       "      <td>-0.662435</td>\n",
       "      <td>-0.019283</td>\n",
       "      <td>-0.057878</td>\n",
       "      <td>0.662435</td>\n",
       "      <td>-0.038573</td>\n",
       "      <td>0.638610</td>\n",
       "      <td>0.253347</td>\n",
       "      <td>-0.213715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24068</th>\n",
       "      <td>chrX</td>\n",
       "      <td>156020825</td>\n",
       "      <td>156020826</td>\n",
       "      <td>ENSG00000182484.15</td>\n",
       "      <td>0.273309</td>\n",
       "      <td>1.683348</td>\n",
       "      <td>-2.160044</td>\n",
       "      <td>-0.333891</td>\n",
       "      <td>-0.927155</td>\n",
       "      <td>-0.374957</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.761843</td>\n",
       "      <td>-0.354350</td>\n",
       "      <td>2.160044</td>\n",
       "      <td>0.761843</td>\n",
       "      <td>0.194028</td>\n",
       "      <td>0.569185</td>\n",
       "      <td>0.019283</td>\n",
       "      <td>-0.638610</td>\n",
       "      <td>-1.281552</td>\n",
       "      <td>-1.481544</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24069 rows × 133 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       #chr      start        end             gene_id  GTEX-11NUK  GTEX-11ONC  \\\n",
       "0      chr1      29552      29553   ENSG00000227232.5   -1.159742   -0.736316   \n",
       "1      chr1      62947      62948   ENSG00000240361.1    0.135385   -0.546656   \n",
       "2      chr1      69090      69091   ENSG00000186092.4    0.233485    2.160044   \n",
       "3      chr1     135894     135895   ENSG00000268903.1   -0.213715    0.019283   \n",
       "4      chr1     137964     137965   ENSG00000269981.1   -0.115950   -0.077204   \n",
       "...     ...        ...        ...                 ...         ...         ...   \n",
       "24064  chrX  155466539  155466540   ENSG00000224533.4   -0.736316   -1.218387   \n",
       "24065  chrX  155669943  155669944  ENSG00000185973.10    1.020076   -1.198380   \n",
       "24066  chrX  155767811  155767812  ENSG00000168939.11    1.053073   -0.761843   \n",
       "24067  chrX  155881292  155881293  ENSG00000124333.15    0.736316   -0.135385   \n",
       "24068  chrX  156020825  156020826  ENSG00000182484.15    0.273309    1.683348   \n",
       "\n",
       "       GTEX-11PRG  GTEX-11TTK  GTEX-11ZU8  GTEX-11ZVC  ...  GTEX-WZTO  \\\n",
       "0       -0.174416    0.524401   -1.238894   -0.686643  ...  -0.569185   \n",
       "1        0.459112    0.761843    1.122762    0.736316  ...  -1.326678   \n",
       "2       -1.159742    0.253347    0.761843    0.897915  ...  -0.057878   \n",
       "3       -1.238894    0.957209   -1.541986   -1.374680  ...  -0.354350   \n",
       "4       -0.897915    0.437790   -0.395725   -1.326678  ...  -0.174416   \n",
       "...           ...         ...         ...         ...  ...        ...   \n",
       "24064   -0.524401   -1.218387    1.087257    1.159742  ...   0.841621   \n",
       "24065    0.869424    2.423196   -0.019283    0.416666  ...  -0.333891   \n",
       "24066    1.238894   -1.374680   -0.174416    0.638610  ...   1.541986   \n",
       "24067    0.233485    0.459112    1.541986    0.374957  ...   0.313572   \n",
       "24068   -2.160044   -0.333891   -0.927155   -0.374957  ...  -0.761843   \n",
       "\n",
       "       GTEX-X4EP  GTEX-X4XX  GTEX-X585  GTEX-XLM4  GTEX-Y8DK  GTEX-YFC4  \\\n",
       "0       0.194028   1.281552   1.020076   0.313572   1.122762   0.480644   \n",
       "1      -1.326678  -0.038573  -0.761843  -1.326678  -1.326678  -0.077204   \n",
       "2      -1.608653  -0.213715  -1.608653  -0.233485  -1.608653   0.480644   \n",
       "3      -0.437790   0.927155   1.683348   0.736316   0.194028   0.213715   \n",
       "4       0.841621   0.354350   2.160044   0.135385  -0.502402   0.194028   \n",
       "...          ...        ...        ...        ...        ...        ...   \n",
       "24064   0.313572  -1.218387   0.194028  -1.218387  -1.218387   0.019283   \n",
       "24065   0.615141  -1.768825  -1.281552   0.437790   0.115950  -0.502402   \n",
       "24066  -0.927155   0.761843  -1.159742  -0.213715   0.897915   0.711260   \n",
       "24067  -1.541986  -0.662435  -0.019283  -0.057878   0.662435  -0.038573   \n",
       "24068  -0.354350   2.160044   0.761843   0.194028   0.569185   0.019283   \n",
       "\n",
       "       GTEX-ZAB4  GTEX-ZAK1  GTEX-ZVZQ  \n",
       "0       1.159742  -1.326678  -0.333891  \n",
       "1      -1.326678  -0.615141   0.057878  \n",
       "2      -0.686643  -1.608653  -0.592007  \n",
       "3       0.135385  -0.416666  -0.592007  \n",
       "4      -0.761843  -0.333891  -0.293381  \n",
       "...          ...        ...        ...  \n",
       "24064  -1.218387  -1.218387   0.686643  \n",
       "24065  -1.159742   0.154871   1.426077  \n",
       "24066  -0.253347  -1.281552   1.768825  \n",
       "24067   0.638610   0.253347  -0.213715  \n",
       "24068  -0.638610  -1.281552  -1.481544  \n",
       "\n",
       "[24069 rows x 133 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GTEx_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "1ab611b7-4a1b-47c0-9a85-ae28102413e7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 14796 overlapped gene_id between our bed.gz and GTEx bed.gz\n",
      "    - 14796 of them have the same positions\n"
     ]
    }
   ],
   "source": [
    "overlapped_gene_id = sorted(list(set(our_bed['gene_id']) & set(GTEx_bed['gene_id'])))\n",
    "print(f'There are {len(overlapped_gene_id)} overlapped gene_id between our bed.gz and GTEx bed.gz')\n",
    "t = 0\n",
    "for i in range(len(overlapped_gene_id)):\n",
    "    trip1 = our_bed.iloc[our_bed['gene_id'].tolist().index(overlapped_gene_id[n]), :][:3]\n",
    "    trip2 = GTEx_bed.iloc[GTEx_bed['gene_id'].tolist().index(overlapped_gene_id[n]), :][:3]\n",
    "    if sum(trip1==trip2)==3:\n",
    "        t = t+1\n",
    "print(f'    - {t} of them have the same positions')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9167bf34-64e3-4059-aae1-7f66644aa9be",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Allen_project_cuda",
   "language": "python",
   "name": "allen_project_cuda"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
